analytics/services/submission_analytics.py

```diff
@@ -1,22 +1,229 @@
+            num_attempts = curr_uqj_submissions.count()
+            is_correct = False
+            for item in curr_uqj_submissions:
+                if item.is_correct is True:
+                    is_correct = True
+                    break
+
+            if isinstance(submission, JavaSubmission):
+                try:
+                    sub_analytics = JavaSubmissionAnalytics.objects.get(submission=submission)
+                except JavaSubmissionAnalytics.DoesNotExist:
+                    ans = submission.answer_files
+                    sub_analytics_dict = SubmissionAnalyticsObj(ans)
+                    user_obj = MyUser.objects.get(pk=submission.user.pk)
+
+                    JavaSubmissionAnalytics.objects.create(uqj=submission.uqj, submission=submission,
+                                                           question=submission.question,
+                                                           event=submission.question.event, user_id=submission.user,
+                                                           first_name=user_obj.first_name, last_name=user_obj.last_name,
+                                                           ans_file=ans, time_spent=time_spent,
+                                                           num_attempts=num_attempts, is_correct=is_correct,
+                                                           lines=sub_analytics_dict.lines,
+                                                           blank_lines=sub_analytics_dict.blank_lines,
+                                                           comment_lines=sub_analytics_dict.comment_lines,
+                                                           import_lines=sub_analytics_dict.imported_lines,
+                                                           cc=sub_analytics_dict.cc,
+                                                           method=sub_analytics_dict.method,
+                                                           operator=sub_analytics_dict.operator,
+                                                           operand=sub_analytics_dict.operand,
+                                                           unique_operator=sub_analytics_dict.unique_operator,
+                                                           unique_operand=sub_analytics_dict.unique_operand,
+                                                           vocab=sub_analytics_dict.vocab,
+                                                           size=sub_analytics_dict.size, vol=sub_analytics_dict.vol,
+                                                           difficulty=sub_analytics_dict.difficulty,
+                                                           effort=sub_analytics_dict.effort,
+                                                           error=sub_analytics_dict.error,
+                                                           test_time=sub_analytics_dict.test_time)
```

Comment:
Find a way to avoid these huge blocks. Maybe a constructor for the model would be useful.

Reply:
I tried with a constructor but doesn't seem like this could be avoided, but I figured out that there are some redundancy in this file so I deleted a lot of code.